DemoApp/Sources/Components/Reactions/ReactionsHelper.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  ReactionsHelper.swift
+//  VideoDogfooding
+//
+//  Created by Martin Mitrevski on 28.2.23.
+//
+
+import Foundation
+import SwiftUI
+import StreamVideo
+import AVFoundation
+
+@MainActor
+final class ReactionsHelper: ObservableObject {
+
+    @Injected(\.streamVideo) var streamVideo
+
+    var player: AVAudioPlayer?
+
+    private var reactionsTask: Task<Any, Error>?
+
+    @Published var reactionsShown = false
+    @Published var showFireworks = false
+    @Published var availableReactions: [Reaction] = [
+        .fireworks,
+        .raiseHand,
+        .lowerHand,
+        .like,
+        .dislike,
+        .heart,
+        .smile
+    ]
+
+    @Published var activeReactions: [String: [Reaction]] = [:]
+
+    var call: Call? {
+        didSet {
+            subscribeToReactionEvents()
+        }
+    }
+
+    init() {}
+
+    func send(reaction: Reaction) {
+        Task {
+            try await call?.sendReaction(
+                type: reaction.id.rawValue,
+                custom: [
+                    "id": .string(reaction.id.rawValue),
+                    "duration": .number(reaction.duration ?? 0),
+                    "userSpecific": .bool(reaction.userSpecific),
+                    "isReverted": .bool(shouldRevert(reaction: reaction))
+                ],
+                emojiCode: reaction.id.rawValue
+            )
+        }
+    }
+
+    func shouldRevert(reaction: Reaction) -> Bool {
+        reaction.id == .raiseHand && activeReactions[streamVideo.user.id]?.first { $0.id == .raiseHand } != nil
+    }
+
+    func removeRaisedHand(from userId: String) {
+        unregister(reaction: .raiseHand, for: userId)
+    }
+
+    private func subscribeToReactionEvents() {
+        guard let call else {
+            reactionsTask?.cancel()
+            return
+        }
+        reactionsTask = Task {
+            for await event in call.subscribe(for: CallReactionEvent.self) {
+                guard
+                    let reaction = reaction(for: event)
+                else {
+                    continue
+                }
+                handleReaction(reaction, from: event.reaction.user.toUser)
+            }
+            return
+        }
+    }
+
+    private func reaction(for event: CallReactionEvent) -> Reaction? {
+        guard let emojiCode = event.reaction.emojiCode else {
+            return nil
+        }
+
+        if let availableReaction = availableReactions.first(where: { $0.id.rawValue == emojiCode }) {
+            return availableReaction
+        } else {
+            return nil
+        }
+    }
+
+    private func handleReaction(_ reaction: Reaction, from user: User) {
+        switch reaction.id {
+        case .fireworks:
+            handleFireworksReaction(reaction, from: user)
+        case .raiseHand:
+            handleRaiseHandReaction(reaction, from: user)
+        case .lowerHand:
+            break
```

Comment:
shouldn't we reverse the reaction here?

Reply:
The `lowerHand` reaction never leaves the device. On the API call we map it to a raise hand reaction. This is to achieve consistent communication with the other clients. We will never receive a `lowerHand` reaction. It will be a `raiseHand`